daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ZonedTextParsers.scala

```diff
@@ -17,14 +17,110 @@
 
 package org.apache.daffodil.processors.parsers
 
+import org.apache.daffodil.schema.annotation.props.gen.TextZonedSignStyle
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.exceptions.UnsuppressableException
+import org.apache.daffodil.util.DecimalUtils
+import org.apache.daffodil.util.DecimalUtils.OverpunchLocation
+import java.text.ParsePosition
+import java.lang.{ Number => JNumber }
+import org.apache.daffodil.infoset.DISimple
 import org.apache.daffodil.processors.ElementRuntimeData
+import org.apache.daffodil.processors.Success
+import org.apache.daffodil.processors.TermRuntimeData
 
-class ZonedTextNumberParser(override val context: ElementRuntimeData)
-  extends TextPrimParser {
+case class ConvertZonedCombinatorParser(
+  rd: TermRuntimeData,
+  valueParser: Parser,
+  converterParser: Parser)
+  extends CombinatorParser(rd) {
 
   override lazy val runtimeDependencies = Nil
 
+  override lazy val childProcessors = Seq(valueParser, converterParser)
+
   def parse(start: PState): Unit = {
-    context.notYetImplemented("Zoned Numbers")
+    valueParser.parse1(start)
+    if (start.processorStatus eq Success) {
+      converterParser.parse1(start)
+    }
+  }
+}
+
+case class ConvertZonedNumberParser[S](
+  helper: ConvertTextNumberParserUnparserHelperBase[S],
+  pattern: String,
+  nff: NumberFormatFactoryBase[S],
+  zonedSignStyle: TextZonedSignStyle,
+  override val context: ElementRuntimeData) extends TextPrimParser {
+  override lazy val runtimeDependencies = Nil
+
+  override def toString = "to(xs:" + helper.xsdType + ")"
+
+  def parse(start: PState): Unit = {
+    val node: DISimple = start.simpleElement
+    val str = node.dataValueAsString
+
+    Assert.invariant(str != null) // worst case it should be empty string. But not null.
+    if (str == "") {
+      PE(start, "Convert to %s (for xs:%s): Cannot parse number from empty string", helper.prettyType, helper.xsdType)
+      return
+    }
+
+    val opindex = pattern.indexOf('+')
+    val opl = {
+      if (opindex == 0)
+        OverpunchLocation.Start
+      else if (opindex == pattern.length - 1)
+        OverpunchLocation.End
+      else
+        OverpunchLocation.None
+    }
```

Comment:
Same comment as above, this logic can be done in the primitives. The pattern shouldn't be needed, just the overpunch location.